src/sema/functions.rs

```diff
@@ -192,6 +192,15 @@ pub fn contract_function(
                     ));
                     success = false;
                 } else {
+                    if ns.target == Target::Solana {
+                        ns.diagnostics.push(Diagnostic::warning(
+                            *loc,
+                            "Selector (or discriminator) overriding may not have the \
+                              desired effect. The metadata file does not contains the overridden \
+                              selector, so the contract will be unused."
```

Comment:
Needs condition "if function has body"

```suggestion
                              selector, so the contract will not function."
```

Reply:
Please, be more attentive in your suggestions. You have given me this message as a suggestion:

https://github.com/hyperledger/solang/pull/1089#discussion_r1044441794

Reply:
I originally suggested `unusable`. You changed this to `unused`, which is wrong.

Reply:
Sorry, that was a typing mistake of mine.